pkg/kubectl/cmd/drain.go

```diff
@@ -282,16 +283,28 @@ func (o *DrainOptions) getController(sr *api.SerializedReference) (interface{},
 	return nil, fmt.Errorf("Unknown controller kind %q", sr.Reference.Kind)
 }
 
+func newSerializedControllerRef(pod api.Pod) *api.SerializedReference {
+	return &api.SerializedReference{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "SerializedReference",
+		},
+		Reference: api.ObjectReference{ //currently only set based on first controller
+			APIVersion: pod.GetOwnerReferences()[0].APIVersion,
+			Kind:       pod.GetOwnerReferences()[0].Kind,
+			Name:       pod.GetOwnerReferences()[0].Name,
+			Namespace:  pod.GetObjectMeta().GetNamespace(),
+		},
+	}
+}
+
 func (o *DrainOptions) getPodCreator(pod api.Pod) (*api.SerializedReference, error) {
-	creatorRef, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
-	if !found {
+	if controller.GetControllerOf(pod.GetObjectMeta()) == nil {
```

Comment:
It should work to just do `GetControllerOf(pod)`.

Also, keep the returned value since it will already contain what you need, and you won't need to use `newSerializedControllerRef()` to pull the same info out of `pod` again.

```go
controllerRef := controller.GetControllerOf(pod)
if controllerRef == nil {
  return nil, nil
}
```